src/zkevm_specs/evm/execution/block_coinbase.py

```diff
@@ -0,0 +1,19 @@
+from ..instruction import Instruction, Transition
+from ..table import BlockContextFieldTag
+from ..opcode import Opcode
+
+
+def coinbase(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+    instruction.constrain_equal(opcode, Opcode.COINBASE)
+    address = instruction.stack_push()
+    # in real circuit also check address raw data is 160 bit length (20 bytes)
+    # check block table for coinbase address
```

Comment:
I assume the block table will somehow be built by verifier (or at least fed into circuit as public input somewhere), so I think EVM circuit could take that it fits in 160 bits.

Reply:
agree with that, in circuit already use `Address` type to represent it.